### ISSUE TITLE
Bug in coef

### ISSUE THREAD

**Issue by bommert** [hazimehh/L0Learn#34]
Hi,

in a real data situation I ran across the error:

```
Error in object$beta[[gammaindex]][, indices, drop = FALSE] : 
  incorrect number of dimensions
```

It can be reproduced with this example:

```
set.seed(1)

X = matrix(rnorm(1000), ncol = 20)
y = rnorm(50)

fit = L0Learn.fit(X, y, penalty = "L0", maxSuppSize = 2)

l = min(fit$lambda[[1]])
coef(fit, l)
```

**Comment by hazimehh** [comment]
Thanks for reporting this bug! In the case of the real dataset was there a single solution in the path?

For some reason, when there's a single solution in the path, the package we're using to interface C++ with R is storing fit$beta[[1]] as a dense vector. However, fit$beta[[1]] is supposed to be a sparse vector -- coef expects sparse vectors. I'll fix this and update the package later today

**Comment by hazimehh** [comment]
The bug is now fixed in the latest version (1.0.9). This can be installed directly from Github using
```{r}
library(devtools)
install_github("hazimehh/L0Learn")
```
The latest version should be available on CRAN in the next few days.
